controller/uninstall_controller.go

```diff
@@ -382,6 +382,46 @@ func (c *UninstallController) deleteCRDs() (bool, error) {
 		return true, c.deleteReplicas(replicas)
 	}
 
+	// Unset backup target URL to prevent the remote backup target
+	// backup config and data, backup volume config be deleted
+	// during uninstall process
+	backupTargets, err := c.ds.ListBackupTargets()
+	if err != nil {
+		return true, err
+	}
+	for btName, bt := range backupTargets {
+		if bt.Spec.BackupTargetURL == "" {
+			delete(backupTargets, btName)
+		}
+	}
+	if len(backupTargets) > 0 {
+		return true, c.unsetBackupTargetsURL(backupTargets)
+	}
+
+	// Delete Backups before deletes engine images
+	if backups, err := c.ds.ListBackups(); err != nil {
+		return true, err
+	} else if len(backups) > 0 {
+		c.logger.Infof("Found %d backups remaining", len(backups))
+		return true, c.deleteBackups(backups)
+	}
+
+	// Delete BackupVolumes before deletes engine images
+	if backupVolumes, err := c.ds.ListBackupVolumes(); err != nil {
+		return true, err
+	} else if len(backupVolumes) > 0 {
+		c.logger.Infof("Found %d backupvolumes remaining", len(backupVolumes))
+		return true, c.deleteBackupVolumes(backupVolumes)
+	}
+
+	// Delete BackupTargets before deletes engine images
+	if backupTargets, err := c.ds.ListBackupTargets(); err != nil {
+		return true, err
+	} else if len(backupTargets) > 0 {
+		c.logger.Infof("Found %d backuptargets remaining", len(backupTargets))
+		return true, c.deleteBackupTargets(backupTargets)
+	}
```

Comment:
If deletes BackupTarget CR first, it would cause backup_volume_controller and backup_controller can't go into deletion code because we check the default BackupTarget CR at these two controllers first.